### ISSUE TITLE
Discuss Semantic visual supported odometry

### ISSUE THREAD

**Issue by AlejandroSilvestri** [yiakwy/SEMANTIC_VISUAL_SUPPORTED_ODEMETRY#1]
Hi @yiakwy, good job!

It's not an easy task to build a new VO.  I can see you are blending deep learning, one step toward Spatial AI (this is a term I read Davidson is installing to name the new visual slam generation with deep learning).

I opened this so called issue, inviting all those who want to join and discuss about this novel VO.

I believe Python is the best choice to deal with deep learning.

**Comment by yiakwy** [comment]
Thank you @AlejandroSilvestri for your support! I feel to be energised to push things forward. Any suggestions are welcome, the project itself is under evolving, I need more help.

**Comment by AlejandroSilvestri** [comment]
@yiakwy 

I saw many visual slam projects, and it seems to me those with good installation instructions were forked and cited more often than other.  It is of great help if you always indicate if this is a work in progress, if the actual version is usable, and what to expect from it.  A YouTube video always helps.

**Comment by yiakwy** [comment]
@AlejandroSilvestri Sorry for the delay of reply. I am currently working migrating cpp modules and plant to push them the next month. I understand that the installation is not very neat and I need to improve them.

First #2 gives me a good hint on improving portability of the system by using docker. I am doing it now. I just figure out a plausible road map for the next steps.

## Proposed roadmap

#### 1th Priority
Make a dockerfile and improve the installation system. How do you @AlejandroSilvestri about it?

#### 2nd Priority
Migrating cpp version of the system (I am currently working on it) and replace different models to make the job with a complete report (either good or bad).

**Comment by AlejandroSilvestri** [comment]
Yes!  I believe containers are perfect encapsulation for this type of work demanding a lot of dependencies.
Docker is perfect for this.
